src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts

```diff
@@ -54,9 +69,100 @@ export class PowerShellProjectCreateStep extends ScriptProjectCreateStep {
             await fse.writeFile(profileps1Path, profileps1);
         }
 
+        const majorVersion: number | undefined = await this.getLatestAzModuleMajorVersion(progress);
+        if (majorVersion !== undefined) {
+            progress.report({
+                message: localize('successfullyConnected', 'Successfully retrieved {0} information from PowerShell Gallery', this.azModuleName)
+            });
+        } else {
+            progress.report({
+                message: localize('failedToConnect', 'Failed to get {0} module version. Edit the requirements.psd1 file when the powershellgallery.com is accessible.', this.azModuleName)
+            });
+        }
+
         const requirementspsd1Path: string = path.join(context.projectPath, requirementspsd1FileName);
         if (await confirmOverwriteFile(requirementspsd1Path)) {
-            await fse.writeFile(requirementspsd1Path, requirementspsd1);
+            if (majorVersion !== undefined) {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1(majorVersion));
+            } else {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1Offine);
+            }
+        }
+    }
+
+    private async getLatestAzModuleMajorVersion(progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<number | undefined> {
```

Comment:
Please wrap all of this in a try/catch, not just the `parseInt` part. We don't want any of this to block creating a project.

In addition please rename `getLatestAzModuleMajorVersion` to `tryGetLatestAzModuleMajorVersion`. The `try` at the front is a common naming pattern we use for methods that return undefined instead of throwing an exception